objectnode/api_middleware.go

```diff
@@ -84,6 +84,19 @@ func (o *ObjectNode) traceMiddleware(next http.Handler) http.Handler {
 			_ = AccessDenied.ServeResponse(w, r)
 		}
 
+		// failed request monitor
+		statusCode := GetStatusCodeFromContext(r)
+		var metricName string
+		var monitorStatusCodes = []string{"400", "403", "500"}
+		for _, monitorStatusCode := range monitorStatusCodes {
+			if monitorStatusCode == statusCode {
+				metricName = fmt.Sprintf("failed_%v", statusCode)
+				continue
+			}
+		}
+		failedMetric := exporter.NewTPCnt(metricName)
+		defer failedMetric.Set(err)
```

Comment:
There is a logic defect here, the current logic does not meet the expected state will be recorded with an empty key.